File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxi=-9999999,n=nums.length,sum=0;
       for(let i=0;i<n;i++){
        let cur=nums[i];
        sum=Math.max(cur,sum+cur);
        maxi=Math.max(maxi,sum);
       }
       return maxi; 
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi=INT_MIN,n=nums.size(),sum=0;
       for(int i=0;i<n;i++){
        int cur=nums[i];
        sum=max(cur,sum+cur);
        maxi=max(maxi,sum);
       }
       return maxi;
    }
};